include/exadg/convection_diffusion/driver.cpp

```diff
@@ -188,28 +187,152 @@ Driver<dim, Number>::ale_update() const
   time_int_bdf->ale_update();
 }
 
+template<int dim, typename Number>
+void
+Driver<dim, Number>::mark_cells_coarsening_and_refinement(dealii::Triangulation<dim> & tria,
+                                                          VectorType const & solution) const
+{
+  mark_cells_kelly_error_estimator(tria,
+                                   pde_operator->get_dof_handler(),
+                                   pde_operator->get_constraints(),
+                                   *pde_operator->get_mapping(),
+                                   solution,
+                                   application->get_parameters().degree +
+                                     1 /* n_face_quadrature_points */,
+                                   application->get_parameters().amr_data);
+}
+
+template<int dim, typename Number>
+void
+Driver<dim, Number>::setup_after_coarsening_and_refinement()
+{
+  pde_operator->initialize_dof_handler_and_constraints();
+  pde_operator->setup();
+
+  postprocessor->setup(*pde_operator);
+
+  VectorType const * velocity_ptr = nullptr;
+  VectorType         velocity;
+
+  if(application->get_parameters().problem_type == ProblemType::Unsteady)
```

Comment:
Is this necessary at all? We only run into such functionality during time stepping, right?

Reply:
I guess this depends on the use case: I could imagine a setup where one wants to implement a loop to compute the solution, evaluate some error estimator, and then run again, until a predefined number of refinement steps have been done or the error is below a certain threshold. It would be similar to the loop over refinements we have now, but acting on a different level (where not everything is set up again). It could be also possible to do it on another level of the code, but in general it can make sense to change the mesh also in stationary problems and keep as much of the data alive, in a similar vein as for time-dependent problems.

Reply:
Yes, we are probably more flexible when removing this if-statement here.

Reply:
My logic here was to simply assert everything I did not test such that if we extend in the future we at least know, where certain cases were not really considered yet.
If you check out [the if-else branch for ProblemType::Steady ind Driver::setup()](https://github.com/richardschu/exadg/blob/adaptivity_restructured/include/exadg/convection_diffusion/driver.cpp#L156-L166), we actually need a different setup. But the steady case does not consider AMR yet as of this PR. Three options: i) remove the if and wait till we have to adapt again once we decide to have AMR for the steady case; ii) implement needed setup for the steady case already despite the rest of the PR does not consider `ProblemType::Steady`; iii) leave as is such that we know in the future where we have to adapt.
I vote for iii) here - am I missing something?